[11.0][mig] stock_valued_picking_report > stock_picking_report_valued

luismontalba opened the pull request:
Migration from V10.0 to V11.0
Module renamed to stock_picking_report_valued

Stock Picking Report Valued
=======================
Add amount information to Delivery Slip report. You can select at partner level if picking list report must be valued or not.

Configuration
-----------------

- Go to *Customers > (select one of your choice) > Sales & Purchases*.
- Set *Valued picking* field on.

Usage
--------

To get the stock picking valued report:

- Create a Sale Order with stockable products a *Valued picking* able
   customer.
- Confirm the Sale Order.
- Click on *Deliveries* stat button.
- Go to *Print > Delivery Slip*.

cc @Tecnativa

Commit:
stock_valued_picking_report

Commit:
[MIG] stock_picking_valued_report: Migrating to v8

Bugfixes

Don't show amounts if picking don't belongs to a sale.order

Propused fixes done

Cambios sugeridos No mostrar el albaran valorado si el partner está configurado para que no se valoren

oihane committed:
<stock_valued_picking_report> depends on stock_picking_taxes

Template inherited and translation updates

Readme changed

<stock_valued_picking_report> code formatted

Fixed README.rst

<stock_valued_picking_report> added stock_picking view

<stock_valued_picking_report> valued field outside footer group

<stock_valued_picking_report> Fixed duplicated fields in view

stock_valued_picking_report: show sale prices in picking lines if valued

carlosdauden committed:
[MIG] stock_valued_picking_report: Migration to 9.0

chienandalu committed:
[IMP] stock_valued_picking_report: Tests and comments

chienandalu committed:
[MIG] stock_valued_picking_report: Migration to 10.0

Commit:
[10.0][FIX] stock_valued_picking_report: Fix report (#38)

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
[11.0][MIG] stock_valued_picking_report

pedrobaeza commented on stock_picking_report_valued/__manifest__.py:
```diff
@@ -1,28 +1,26 @@
 
 {
-    "name": "Stock Valued Picking Report",
+    "name": "Stock Picking Report Valued",
```
It was correct before. We can remove `Stock` part if you want.

pedrobaeza commented on stock_picking_report_valued/__manifest__.py:
```diff
@@ -1,28 +1,26 @@
+    "name": "Stock Picking Report Valued",
     "summary": "Adding Valued Picking on Delivery Slip report",
-    "version": "10.0.1.0.1",
+    "version": "11.0.1.0.1",
```
11.0.1.0.0

pedrobaeza commented on stock_picking_report_valued/models/stock_move_line.py:
```diff
@@ -0,0 +1,40 @@
+from odoo import api, fields, models
+
+
+class StockPackOperation(models.Model):
```
Rename to StockMoveLine

pedrobaeza commented on stock_picking_report_valued/models/stock_move_line.py:
```diff
@@ -0,0 +1,40 @@
+    @api.multi
+    @api.depends('sale_line.tax_id')
+    def _compute_sale_tax_description(self):
+        for operation in self:
```
s/operation/line

pedrobaeza commented on stock_picking_report_valued/models/stock_move_line.py:
```diff
@@ -0,0 +1,40 @@
+class StockPackOperation(models.Model):
+    _inherit = "stock.move.line"
+
+    currency_id = fields.Many2one(
```
I think we don't need all this overload in the model. As they're related fields and they are not shown in any view, you can use dot notation for accessing its values in the report.

luismontalba replied:
I've tried to remove it but test is telling me it is necessary. I think the monetary field needs this one as an auxiliary field

pedrobaeza replied:
Well, I'm talking about all the fields, but OK, let's continue this way.

luismontalba commented:
Changes done

rafaelbn commented:
Testing in runbot I find something that maybe should be treated.

In the stock.picking , with all unit reserved but still not validated if you print the Delivery slip you get a PDF valued but incorrect because quantity is 0. Check screenshot,  you see valued with $ but how many? unit(s)

![2018-04-10_22-18-38](https://user-images.githubusercontent.com/8736623/38581561-83a6ed72-3d0d-11e8-887f-f9521f2891f0.png)

:thinking: 

Check this one http://3322527-44-613e08.runbot1.odoo-community.org/web#id=19&view_type=form&model=stock.picking&action=249&active_id=22

rafaelbn reviewed (commented):
Tested with one doubt commented previously

luismontalba commented:
I'll review it

pedrobaeza commented on stock_picking_report_valued/README.rst:
```diff
@@ -0,0 +1,80 @@
+Add amount information to Delivery Slip report.
+You can select at partner level if picking list report must be valued or not.
+If the picking is done it's valued with quantity done, otherwise the picking
+ is valued with reserved quantity.
```
Don't add this space at the beginning

luismontalba replied:
Fixed (commit squashed)

luismontalba commented:
@rafaelbn , now the picking is valued with done quantity when the state is done, and otherwise it's valued with reserved quantity.

carlosdauden commented on stock_picking_report_valued/models/stock_move_line.py:
```diff
@@ -0,0 +1,61 @@
+        for line in self:
+            price = line.sale_price_unit * (1 - (
+                line.sale_discount or 0.0) / 100.0)
+            if line.picking_id.state == 'done':
```
Is needed this condition?
Don't change product_uom_qty  to qty_done when transfer is done?

luismontalba replied:
Well, depending on state of the picking, the quantity to take is in the field product_uom_qty or is in the field qty_done. This is the way I found to value the picking in both states, but any better way is welcome!

pedrobaeza commented:
@carlosdauden what is missing here?

carlosdauden committed:
[IMP] stock_picking_report_valued: Improve code

pedrobaeza committed:
[IMP] stock_picking_report_valued: Perform related & compute with sudo

For avoiding problems if you don't have access to sales orders (stricter
warehouse users, inter-company records...).

chienandalu committed:
[FIX] stock_picking_report_valued: core view change

- This commit
https://github.com/odoo/odoo/commit/41bea4125cc898bbcc7ca14fa5be112512c67d8e
breaked the inheritance.
